### ISSUE TITLE
Incorrect verdict on numpy.clip

### ISSUE THREAD

**Issue by pedro-w** [microsoft/pyright#5678]
**Description**
In VS Code, an error is flagged when the code is (I believe) correct.
`casting` is a valid kwarg for the `numpy.clip` ufunc ([docs](https://numpy.org/doc/stable/reference/ufuncs.html#ufuncs-kwargs))

**Minimal Code Sample**

```python
import numpy

numpy.clip(numpy.zeros(1), 0, 255, casting='safe')
```

    No overloads for "clip" match the provided arguments
     Argument types: (NDArray[float64], Literal[0], Literal[255], Literal['safe'])

**Version**
Version: 1.81.0 (user setup)
Commit: 6445d93c81ebe42c4cbd7a60712e0b17d9463e97
Date: 2023-08-02T12:37:13.485Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

Pylance: v2023.8.20

numpy: 1.25.2
